oldlog.go

```diff
@@ -97,6 +97,19 @@ func SetLogLevel(name, level string) error {
 	return nil
 }
 
+// GetSubsystems returns a slice containing the
+// names of the current loggers
+func GetSubsystems() []string {
+	subs := make([]string, len(loggers))
+
+	i := 0
+	for k := range loggers {
```

Comment:
The solution I posted is efficient (though maybe yours has very slight gains on reslicing costs). We preallocate the buffer (note the extra zero in the `make`) so that the appends don't have to reallocate any memory.